core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -658,16 +659,18 @@ public SqlNode toSql(@Nullable RexProgram program, RexNode rex) {
           final SqlNode expr = toSql(program, referencedExpr);
           sqlIdentifier = new SqlIdentifier(expr.toString(), POS);
           break;
+        case ITEM:
```

Comment:
The switch branch for `ROW` is not correct too. Maybe we should unify them and make this the default behavior?